lib/keys/keys.ts

```diff
@@ -0,0 +1,175 @@
+    const { PrivateKey, ID } = Keys.find(({ Primary }) => Primary === 1) || {};
+    const { KeySalt } = KeySalts.find(({ ID: keySaltID }) => ID === keySaltID) || {};
+
+    // Not verifying that KeySalt exists because of old auth versions.
+    return {
+        PrivateKey,
+        KeySalt
+    };
+};
+
+interface GenerateAddressKeyArguments {
+    email: string;
+    name?: string;
+    passphrase: string;
+    encryptionConfig: EncryptionConfig;
+}
+export const generateAddressKey = async ({
+    email,
+    name = email,
+    passphrase,
+    encryptionConfig
+}: GenerateAddressKeyArguments) => {
+    const { key: privateKey, privateKeyArmored } = await generateKey({
+        userIds: [{ name, email }],
+        passphrase,
+        ...encryptionConfig
+    });
+
+    await privateKey.decrypt(passphrase);
+
+    return { privateKey, privateKeyArmored };
+};
+
+interface ReformatAddressKeyArguments {
+    email: string;
+    name?: string;
+    passphrase: string;
+    privateKey: OpenPGPKey;
+}
+export const reformatAddressKey = async ({
+    email,
+    name = email,
+    passphrase,
+    privateKey: originalKey
+}: ReformatAddressKeyArguments) => {
+    const { key: privateKey, privateKeyArmored } = await reformatKey({
+        userIds: [{ name, email }],
+        passphrase,
+        privateKey: originalKey
+    });
+
+    await privateKey.decrypt(passphrase);
+
+    return { privateKey, privateKeyArmored };
+};
+
+/**
+ * Decrypt an address key token and verify the detached signature.
+ */
+interface DecryptAddressKeyTokenArguments {
+    Token: string;
+    Signature: string;
+    privateKeys: OpenPGPKey | OpenPGPKey[];
+    publicKeys: OpenPGPKey | OpenPGPKey[];
+}
+export const decryptAddressKeyToken = async ({
+    Token,
+    Signature,
+    privateKeys,
+    publicKeys
+}: DecryptAddressKeyTokenArguments) => {
+    const { data: decryptedToken, verified } = await decryptMessage({
+        message: await getMessage(Token),
+        signature: await getSignature(Signature),
+        privateKeys,
+        publicKeys
+    });
+
+    if (verified !== VERIFICATION_STATUS.SIGNED_AND_VALID) {
+        const error = new Error(c('Error').t`Signature verification failed`);
+        error.name = 'SignatureError';
+        throw error;
+    }
+
+    return decryptedToken;
+};
+
+export const splitKeys = (keys: CachedKey[] = []) => {
+    return keys.reduce<{ privateKeys: OpenPGPKey[]; publicKeys: OpenPGPKey[] }>(
+        (acc, { privateKey, publicKey }) => {
+            if (!privateKey || !publicKey) {
+                return acc;
+            }
+            acc.publicKeys.push(publicKey);
+            acc.privateKeys.push(privateKey);
+            return acc;
+        },
+        { publicKeys: [], privateKeys: [] }
+    );
+};
+
+interface GetAddressKeyTokenArguments {
+    Token: string;
+    Signature: string;
+    organizationKey?: OpenPGPKey | void;
```

Comment:
`void` is not something we use to type parameters. It's really only meant to be used as a return type for a procedure - function that returns nothing.